File: Intermediate/Java/problem_20.java
import java.util.Scanner;
public class Main
{
  public static void main (String[]args)
  {
    Scanner sc = new Scanner (System.in);

    int i, fact = 1;
    System.out.println ("Please Enter Number for calculating factorial");
    int number = sc.nextInt ();
      fact = factorial (number);	// function to calculate factorial using Recursive approch.  
      System.out.println ("Factorial of " + number + " is: " + fact);
  }

  private static int factorial (int n)
  {
    if (n == 0)
      return 1;
    else
      return (n * factorial (n - 1));
  }
}

File: Intermediate/C/problem_20.c
// factorial of a number
#include<stdio.h>
int fact(int n)
{ int f=1,i;
  for(i=1;i<=n;i++)
   f=f*i;
   return f;
}
void main()
{int a;
printf("Enter a number:");
scanf("%d",&a);
printf("\n Factorial of %d:%d",a,fact(a));
}
